scripts/inference.py

```diff
@@ -184,6 +193,22 @@ def pad_prompt(prompt, pad_len, pad_token="<unk>"):
 
 ids = prompt1.unsqueeze(0)
 
+if args.compile:
+    print("compiling model")
+    # compiling can make first inference pass slow
+    model = torch.compile(model, mode=args.compile_mode)
+
+# Export and load the model
+if args.export_model:
+    print("Exporting the compiled model...")
+    example_inputs = (ids,)
+    exported_program = export(model, args=example_inputs)
```

Comment:
You may want to check if a exported file already exists here, in that case you can just load and skip export and save.